### PR TITLE
Makeshift Weaponry expansion

### PR DESCRIPTION
**Pull request by ShadeAware**
### CRETH TECH

**Introduction**
Increases the availability, ease of construction, and amount of **Makeshift Weaponry** available to crewmen.
Rather than simply random assortments of bullshit cobbled together to make one random-caliber weapon, you can now find and make _pieces_ of weaponry; combine weaponry pieces together, and you can make specific kinds of weapons, including guns and bombs!
image
image

**How do I craft?**
In absence of a proper crafting menu, I have tried to make the crafting process as easy as possible within the slap-craft framework; all components and steps have descriptions as to what they do and what they need. As a basic rule of thumb, the general crafting process goes something like this:
1. Find a _Gun Chassis_ and secure it with a screwdriver.
2. Find some _Gun Internals_ and install them into the chassis.
3. Install a _Gun Barrel_
4. Install a _Grip_
5. Find and install a _specification part._ Specification parts are unique bits of a gun that decide the final end product!

As a bonus, this PR also includes a **wrench blade**, made by combining two machetes together and welding them, then adding a wrench as a hilt. Its a shoddy claymore, essentially, but hey it looks cool.

### This PR does NOT include mapspawns for the crafting components. Those will be done separately. All components can be made with an Autolathe, though.
image
image
image

### CODE REVIEW

**Comment by ShadeAware** [comment]
GOOD ENOUGH

**Comment by Kaksisilma** [comment]
Gonna be testing this tomorrow and if any balance-related issue comes up I'll let you know @ShadeAware
otherwise looks great.

**Comment by ShadeAware** [comment]
Duly noted, the balance on this is subject to change due to some very rough numbers.

**Comment by ShadeAware** [comment]
No longer good enough because shit was **CHANGED AGAIN**

File: code/modules/projectiles/guns/projectile/zipgun.dm [hunk 1]
```diff
@@ -37,3 +37,123 @@
...
+/obj/item/weapon/gun/projectile/pistol/improvised
+	name = "scrapper-9"
+	desc = "An improvised pistol capable of utilizing standard magazines. Its makeshift nature sacrifices power for speed; shoddy, but useful in a pinch."
+	icon = 'icons/obj/guns/zipgun.dmi'
+	icon_state = "scrapper"
+	caliber = CALIBER_PISTOL_SMALL
+	fire_delay = 5
+	damage_mult = 0.9
+	penetration_mult = 0.8
+	has_safety = FALSE
```
Comment (Kaksisilma) [inline code/modules/projectiles/guns/projectile/zipgun.dm:49]: couple of failed checks here regarding penetration_mult

**Review by Kaksisilma — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/projectile/zipgun.dm (hunk 1 above)
Comment (ShadeAware) [reply code/modules/projectiles/guns/projectile/zipgun.dm:49]: Im well aware

**Comment by ShadeAware** [comment]
that should hopefully be the end of the tomfuckery

**Review by Kaksisilma — APPROVED** [review]
Good to merge?

**Pull request merged**